### PR TITLE
feat(deeplink): add deeplink as new backend

### PR DESCRIPTION
**Pull request by caikun-pjlab**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

add DeepLink as a new backend for InternEvo.

## Modification

add DeepLink as a new backend for InternEvo.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Review by SolenoidWGT — APPROVED** [review]

File: internlm/core/communication/isp.py [hunk 1]
```diff
@@ -16,16 +15,25 @@
 from internlm.utils.common import SchedulerHook, get_current_device
 
 
-@dataclass
 class ISPCommModelConfig:
```
Comment (sunpengsdu) [inline internlm/core/communication/isp.py:18]: 为啥要改这里啊？

File: internlm/core/communication/isp.py [hunk 2]
```diff
@@ -16,16 +15,25 @@
...
     """
     model config for isp communicator.
     """
 
-    dtype: torch.dtype = torch.half
-    device: torch.device = get_current_device()
```
Comment (SolenoidWGT) [inline internlm/core/communication/isp.py:26]: 这里修改是为了防止在NPU任务在import的时候就调用get_current_device()，这时还没有调用torch.set_device，会导致出现device设置错误的问题

File: internlm/core/communication/isp.py (hunk 1 above)
Comment (SolenoidWGT) [reply internlm/core/communication/isp.py:18]: 这里修改是为了防止在NPU任务在import的时候就调用get_current_device()，这时还没有调用torch.set_device，会导致出现device设置错误的问题

File: internlm/model/metrics.py [hunk 3]
```diff
@@ -262,7 +262,7 @@ def __init__(self, device, dp_pg, dataset_types: List[str] = None) -> None:
             self.ds_loss = torch.zeros(self.total_type_count, dtype=torch.float, device=device)
             self.ds_token_num = torch.zeros(self.total_type_count, dtype=torch.float, device=device)
 
-        if gpc.config.use_cuda_flash_attn:
+        if gpc.config.use_cuda_flash_attn and AcceleratorType.GPU == get_accelerator():
```
Comment (sunpengsdu) [inline internlm/model/metrics.py:265]: 似乎应该不用改，use_cuda_flash_attn 就表明了 cuda + fa

Comment (sunpengsdu) [reply internlm/model/metrics.py:265]: 不过没有问题

**Pull request merged**